hack/istio/install-bookinfo-demo.sh

```diff
@@ -359,6 +369,16 @@ $CLIENT_EXE get pods -n ${NAMESPACE}
 if [ "${AMBIENT_ENABLED}" == "true" ]; then
   echo "Sidecar injection was not performed. Ambient support will be enabled."
   ${CLIENT_EXE} label namespace ${NAMESPACE} istio.io/dataplane-mode=ambient
+    # It could also be applied to service account
+    if [ "${WAYPOINT}" == "true" ] && [ "${AMBIENT_ENABLED}" == "true" ]; then
+      # Verify Gateway API
+      echo "Verify if Istio API is installed"
+      $CLIENT_EXE get crd gateways.gateway.networking.k8s.io &> /dev/null || \
+        { $CLIENT_EXE kustomize "github.com/kubernetes-sigs/gateway-api/config/crd?ref=v0.6.2" | $CLIENT_EXE apply -f -; }
+      # Create Waypoint proxy
+      echo "Create Waypoint proxy"
+      ${ISTIOCTL} x waypoint apply -n ${NAMESPACE}
+    fi
```

Comment:
I'll test this shortly, but...

1. You are in an if-stmt that already determined ambient is enabled - no need to check again.
2. Indentation is off by 1.
3. Regarding the message `"Verify if Istio API is installed"` - the code looks like it is looking for the Gateway API and, if it is not, it will install it. So I suggest changing the message to be more clear.

```suggestion
   # It could also be applied to service account
   if [ "${WAYPOINT}" == "true" ]; then
     # Verify Gateway API
     echo "Verifying that Gateway API is installed; if it is not then it will be installed now."
     $CLIENT_EXE get crd gateways.gateway.networking.k8s.io &> /dev/null || \
       { $CLIENT_EXE kustomize "github.com/kubernetes-sigs/gateway-api/config/crd?ref=v0.6.2" | $CLIENT_EXE apply -f -; }
     # Create Waypoint proxy
     echo "Create Waypoint proxy"
     ${ISTIOCTL} x waypoint apply -n ${NAMESPACE}
   fi
```

Reply:
Thanks for the review, @jmazzitelli ! 
I've make a commit regarding indentation - I guess it looks right now?